tools/projmgr/src/ProjMgr.cpp

```diff
@@ -1005,25 +1005,30 @@ bool ProjMgr::RunListToolchains(void) {
     success = false;
   }
 
-  StrSet toolchainsSet;
-  for (const auto& toolchain : toolchains) {
-    string toolchainEntry = toolchain.name + "@" +
-      (toolchain.required.empty() ? toolchain.version : toolchain.required) + "\n";
-    if (m_verbose) {
-      string env = toolchain.version;
-      replace(env.begin(), env.end(), '.', '_');
-      if (!toolchain.root.empty()) {
-        toolchainEntry += "  Environment: " + toolchain.name + "_TOOLCHAIN_" + env + "\n";
-        toolchainEntry += "  Toolchain: " + toolchain.root + "\n";
-      }
-      if (!toolchain.config.empty()) {
-        toolchainEntry += "  Configuration: " + toolchain.config + "\n";
+  if (toolchains.empty()) {
+    cerr << "No valid compiler registered. Consider to define the CMSIS_COMPILER_ROOT environment variable and compiler registration environment variables, format: <GCC|CLANG|AC6|IAR>_TOOLCHAIN_<major>_<minor>_<patch>\n";
```

Comment:
The `CMSIS_COMPILER_ROOT` is optional as documented in the [installation guide](https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/installation.md#environment-variables).
It would be useful to understand in which scenario this is mandatory for you, could you please give us more information on this matter?

Since this regards the generic call of `list toolchains`, without specifying csolution.yml, a more appropriate location to introduce such error message could be here:
https://github.com/Open-CMSIS-Pack/devtools/blob/tools/projmgr/2.4.0/tools/projmgr/src/ProjMgrWorker.cpp#L4050

Other observations:
- Please use `ProjMgrLogger::Error` instead of directly writing to `cerr`.
- For consistency please write error messages in a single sentence without final dot.

Reply:
I know that the `CMSIS_COMPILER_ROOT` is optional, but in some cases, it's an alone csolution binary and in that case, the user needs some hints to what is needed to get it running. Maybe the distribution was missing some files, maybe there is no compatible version of the files in the `etc` directory, regardless, there should be some help to get it working.

I was considering putting the block in the ProjMgrWorker class, but as there are other output in the ProjMgr class in this call hierarchy, I though it made more sense to have it there. Do you still want me to move it?

Also, the reason to call `cerr` was that `cout` is used in the other branch of the function in ProjMgr class.

Let me know how you want to proceed.
Once I have the reply to the above statements, I will try to rephrase the sentences a bit.

Reply:
> I know that the CMSIS_COMPILER_ROOT is optional, but in some cases, it's an alone csolution binary and in that case, the user needs some hints to what is needed to get it running. Maybe the distribution was missing some files, maybe there is no compatible version of the files in the etc directory, regardless, there should be some help to get it working.

Then we would need more separated warnings/errors in different places TBD:
- no toolchain environment variables
- `<cmsis-toolbox>/etc` directory does not exist and `CMSIS_COMPILER_ROOT` was not set or is invalid
- no compatible toolchain config files

It's fine for me if you want to proceed with this PR without addressing those now but in my opinion the most useful would be to inform about a missing `etc` directory that is a symptom of a malformed installation. 
(@ReinhardKeil @jkrech please advise)

> I was considering putting the block in the ProjMgrWorker class, but as there are other output in the ProjMgr class in this call hierarchy, I though it made more sense to have it there. Do you still want me to move it?

Yes please.

> Also, the reason to call cerr was that cout is used in the other branch of the function in ProjMgr class.

`cout` is used for outputing results of `list` commands.
All messages (info, warning, error, debug) should use `ProjMgrLogger`.

Thanks!

Reply:
`csolution` is not expected to work without  `<cmsis-toolbox>/etc` being present. Where `<cmsis-toolbox>/bin` specifies the path of the running csolution binary itself.
In the etc/ directory `csolution` expects schema files and the `global.generator.yml`.

=> error message if `../etc` does not exist
=> error message in schema check when a schema file is missing from `<cmsis-toolbox>/etc`
=> error when not finding a matching toolchain configuration file `<toolchain>.<minVersion>.cmake`. 
IF $CMSIS_COMPILER_ROOT is set, then the message reads: not found in `$CMSIS_COMPILER_ROOT: <cmsis_compiler_root_value>`
ELSE
the message reads: not found in `<cmsis-toolbox>/etc: <cmsis-toolbox-value>/etc`

This looks like three different error messages in three different points in the execution.

Reply:
Sorry, but I will not have time to do this in the near future.
Is this something that Arm can complete or should it be postponed?

Reply:
@Torbjorn-Svensson is that something you would have time to work on in the next couple of weeks? 
If so, please consider the message suggested in the last comment of #1728.